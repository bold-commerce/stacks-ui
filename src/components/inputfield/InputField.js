import React, { Component } from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';

import Field from '../field/Field';
import Input from '../input/Input';
import { generateUniqueId } from '../../utils/generateUniqueId';

import './InputField.css';

class InputField extends Component {
  constructor(props) {
    super(props);
    this.forId = `stxField${generateUniqueId()}`;
  }

  render() {
    const { className, disabled, helpText, id, label, messageText, messageType, readOnly, ariaLive, ...otherProps } = this.props;
    const classNames = cn([
      'stx-field--with-input',
      className,
    ]);

    return (
      <Field
        label={label}
        htmlFor={id || this.forId}
        helpText={helpText}
        className={classNames}
        messageType={messageType}
        messageText={messageText}
        disabled={disabled}
        readOnly={readOnly}
        ariaLive={ariaLive}
      >
        <Input
          {...otherProps}
          id={id || this.forId}
          messageType={messageType}
          disabled={disabled}
          readOnly={readOnly}
        />
      </Field>
    );
  }
}

InputField.defaultProps = {};

InputField.propTypes = {
  /** HTML ID attribute for the input element */
  id: PropTypes.string,
  /** The title or name of the field */
  label: PropTypes.node,
  /** Additional helpful text to describe how to use the field */
  helpText: PropTypes.node,
  /** Optional className for the Input component */
  className: PropTypes.string,
  messageType: PropTypes.oneOf(['success', 'warning', 'alert', '']),
  /** The text that should appear as a message */
  messageText: PropTypes.string,
  /** The aria-live attribute value that will be set on the message if one is given */
  ariaLive: PropTypes.oneOf(['assertive', 'polite']),
  /** Whether or not the field is disabled */
  disabled: PropTypes.bool,
  /** Whether or not the field is read only */
  readOnly: PropTypes.bool,
};

export default InputField;
