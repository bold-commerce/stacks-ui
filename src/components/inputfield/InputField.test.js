/* global describe, test, expect */
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import React from 'react';
import InputField from './InputField';

describe('InputField', () => {
  test('InputField renders with basic props', () => {
    const onFocus = () => 'onFocus';
    const onBlur = () => 'onBlur';
    const onChange = () => 'onChange';
    const onKeyUp = () => 'onKeyUp';
    const onKeyDown = () => 'onKeyDown';

    const output = shallow(<InputField
      id='test-id'
      className='html-class'
      name='name'
      value='test-string'
      placeholder='placeholder'
      data-additional-prop='additional-prop-test'
      type='text'
      onFocus={onFocus}
      onBlur={onBlur}
      onChange={onChange}
      onKeyUp={onKeyUp}
      onKeyDown={onKeyDown}
      ariaLive='assertive'
      selectOnFocus
    />);

    expect(toJson(output)).toMatchSnapshot();
    expect(output.props().children.props.onChange).toEqual(onChange);
    expect(output.props().children.props.onKeyUp).toEqual(onKeyUp);
    expect(output.props().children.props.onKeyDown).toEqual(onKeyDown);
  });

  test('InputField renders without any props', () => {
    const output = shallow(<InputField />);
    expect(toJson(output)).toMatchSnapshot();
  });

  test('InputField renders with a prefix', () => {
    const output = shallow(<InputField prefix='-' />);
    expect(toJson(output)).toMatchSnapshot();
  });

  test('InputField renders with a suffix', () => {
    const output = shallow(<InputField suffix='px' />);
    expect(toJson(output)).toMatchSnapshot();
  });

  test('InputField renders with type text', () => {
    const output = shallow(<InputField type='text' />);
    expect(toJson(output)).toMatchSnapshot();
  });

  test('InputField renders with type date', () => {
    const output = shallow(<InputField type='date' />);
    expect(toJson(output)).toMatchSnapshot();
  });

  test('InputField renders with type datetime-local', () => {
    const output = shallow(<InputField type='datetime-local' />);
    expect(toJson(output)).toMatchSnapshot();
  });

  test('InputField renders with type time', () => {
    const output = shallow(<InputField type='time' />);
    expect(toJson(output)).toMatchSnapshot();
  });

  test('InputField renders with type search', () => {
    const output = shallow(<InputField type='search' />);
    expect(toJson(output)).toMatchSnapshot();
  });

  test('InputField renders with type number', () => {
    const output = shallow(<InputField type='number' />);
    expect(toJson(output)).toMatchSnapshot();
  });

  test('InputField renders with type tel', () => {
    const output = shallow(<InputField type='tel' />);
    expect(toJson(output)).toMatchSnapshot();
  });

  test('InputField renders with type email', () => {
    const output = shallow(<InputField type='email' />);
    expect(toJson(output)).toMatchSnapshot();
  });

  test('InputField renders with type url', () => {
    const output = shallow(<InputField type='url' />);
    expect(toJson(output)).toMatchSnapshot();
  });

  test('InputField renders with type password', () => {
    const output = shallow(<InputField type='password' />);
    expect(toJson(output)).toMatchSnapshot();
  });

  test('InputField renders with type file', () => {
    const output = shallow(<InputField type='file' />);
    expect(toJson(output)).toMatchSnapshot();
  });

  test('InputField renders with size small', () => {
    const output = shallow(<InputField size='small' />);
    expect(toJson(output)).toMatchSnapshot();
  });

  test('InputField renders with a message type of alert', () => {
    const output = shallow(<InputField messageType='alert' />);
    expect(toJson(output)).toMatchSnapshot();
  });

  test('InputField renders with a message type of warning', () => {
    const output = shallow(<InputField messageType='warning' />);
    expect(toJson(output)).toMatchSnapshot();
  });

  test('InputField renders with a message type of success', () => {
    const output = shallow(<InputField messageType='success' />);
    expect(toJson(output)).toMatchSnapshot();
  });

  test('InputField renders with a min and max value', () => {
    const output = shallow(<InputField min={1} max={5} />);
    expect(toJson(output)).toMatchSnapshot();
  });

  test('InputField renders with a min and max length', () => {
    const output = shallow(<InputField minLength={1} maxLength={5} />);
    expect(toJson(output)).toMatchSnapshot();
  });

  test('InputField renders with a step value', () => {
    const output = shallow(<InputField step={5} />);
    expect(toJson(output)).toMatchSnapshot();
  });

  test('InputField renders with a tabIndex value', () => {
    const output = shallow(<InputField tabIndex={-5} />);
    expect(toJson(output)).toMatchSnapshot();
  });

  test('InputField renders with a autoComplete value', () => {
    const output = shallow(<InputField autoComplete='off' />);
    expect(toJson(output)).toMatchSnapshot();
  });

  test('InputField render disabled when disabled is true', () => {
    const output = shallow(<InputField disabled />);
    expect(toJson(output)).toMatchSnapshot();
  });

  test('InputField render readOnly when readOnly is true', () => {
    const output = shallow(<InputField readOnly />);
    expect(toJson(output)).toMatchSnapshot();
  });

  test('InputField render with a loading spinner when loading is true', () => {
    const output = shallow(<InputField loading />);
    expect(toJson(output)).toMatchSnapshot();
  });
});
