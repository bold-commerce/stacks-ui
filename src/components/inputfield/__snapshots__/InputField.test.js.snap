// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField InputField render disabled when disabled is true 1`] = `
<Field
  className="stx-field--with-input"
  disabled={true}
  htmlFor="stxField25"
>
  <Input
    disabled={true}
    id="stxField25"
    selectOnFocus={false}
    type="text"
  />
</Field>
`;

exports[`InputField InputField render readOnly when readOnly is true 1`] = `
<Field
  className="stx-field--with-input"
  htmlFor="stxField26"
  readOnly={true}
>
  <Input
    id="stxField26"
    readOnly={true}
    selectOnFocus={false}
    type="text"
  />
</Field>
`;

exports[`InputField InputField render with a loading spinner when loading is true 1`] = `
<Field
  className="stx-field--with-input"
  htmlFor="stxField27"
>
  <Input
    id="stxField27"
    loading={true}
    selectOnFocus={false}
    type="text"
  />
</Field>
`;

exports[`InputField InputField renders with a autoComplete value 1`] = `
<Field
  className="stx-field--with-input"
  htmlFor="stxField24"
>
  <Input
    autoComplete="off"
    id="stxField24"
    selectOnFocus={false}
    type="text"
  />
</Field>
`;

exports[`InputField InputField renders with a message type of alert 1`] = `
<Field
  className="stx-field--with-input"
  htmlFor="stxField17"
  messageType="alert"
>
  <Input
    id="stxField17"
    messageType="alert"
    selectOnFocus={false}
    type="text"
  />
</Field>
`;

exports[`InputField InputField renders with a message type of success 1`] = `
<Field
  className="stx-field--with-input"
  htmlFor="stxField19"
  messageType="success"
>
  <Input
    id="stxField19"
    messageType="success"
    selectOnFocus={false}
    type="text"
  />
</Field>
`;

exports[`InputField InputField renders with a message type of warning 1`] = `
<Field
  className="stx-field--with-input"
  htmlFor="stxField18"
  messageType="warning"
>
  <Input
    id="stxField18"
    messageType="warning"
    selectOnFocus={false}
    type="text"
  />
</Field>
`;

exports[`InputField InputField renders with a min and max length 1`] = `
<Field
  className="stx-field--with-input"
  htmlFor="stxField21"
>
  <Input
    id="stxField21"
    maxLength={5}
    minLength={1}
    selectOnFocus={false}
    type="text"
  />
</Field>
`;

exports[`InputField InputField renders with a min and max value 1`] = `
<Field
  className="stx-field--with-input"
  htmlFor="stxField20"
>
  <Input
    id="stxField20"
    max={5}
    min={1}
    selectOnFocus={false}
    type="text"
  />
</Field>
`;

exports[`InputField InputField renders with a prefix 1`] = `
<Field
  className="stx-field--with-input"
  htmlFor="stxField3"
>
  <Input
    id="stxField3"
    prefix="-"
    selectOnFocus={false}
    type="text"
  />
</Field>
`;

exports[`InputField InputField renders with a step value 1`] = `
<Field
  className="stx-field--with-input"
  htmlFor="stxField22"
>
  <Input
    id="stxField22"
    selectOnFocus={false}
    step={5}
    type="text"
  />
</Field>
`;

exports[`InputField InputField renders with a suffix 1`] = `
<Field
  className="stx-field--with-input"
  htmlFor="stxField4"
>
  <Input
    id="stxField4"
    selectOnFocus={false}
    suffix="px"
    type="text"
  />
</Field>
`;

exports[`InputField InputField renders with a tabIndex value 1`] = `
<Field
  className="stx-field--with-input"
  htmlFor="stxField23"
>
  <Input
    id="stxField23"
    selectOnFocus={false}
    tabIndex={-5}
    type="text"
  />
</Field>
`;

exports[`InputField InputField renders with basic props 1`] = `
<Field
  ariaLive="assertive"
  className="stx-field--with-input html-class"
  htmlFor="test-id"
>
  <Input
    data-additional-prop="additional-prop-test"
    id="test-id"
    name="name"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    placeholder="placeholder"
    selectOnFocus={true}
    type="text"
    value="test-string"
  />
</Field>
`;

exports[`InputField InputField renders with size small 1`] = `
<Field
  className="stx-field--with-input"
  htmlFor="stxField16"
>
  <Input
    id="stxField16"
    selectOnFocus={false}
    size="small"
    type="text"
  />
</Field>
`;

exports[`InputField InputField renders with type date 1`] = `
<Field
  className="stx-field--with-input"
  htmlFor="stxField6"
>
  <Input
    id="stxField6"
    selectOnFocus={false}
    type="date"
  />
</Field>
`;

exports[`InputField InputField renders with type datetime-local 1`] = `
<Field
  className="stx-field--with-input"
  htmlFor="stxField7"
>
  <Input
    id="stxField7"
    selectOnFocus={false}
    type="datetime-local"
  />
</Field>
`;

exports[`InputField InputField renders with type email 1`] = `
<Field
  className="stx-field--with-input"
  htmlFor="stxField12"
>
  <Input
    id="stxField12"
    selectOnFocus={false}
    type="email"
  />
</Field>
`;

exports[`InputField InputField renders with type file 1`] = `
<Field
  className="stx-field--with-input"
  htmlFor="stxField15"
>
  <Input
    id="stxField15"
    selectOnFocus={false}
    type="file"
  />
</Field>
`;

exports[`InputField InputField renders with type number 1`] = `
<Field
  className="stx-field--with-input"
  htmlFor="stxField10"
>
  <Input
    id="stxField10"
    selectOnFocus={false}
    type="number"
  />
</Field>
`;

exports[`InputField InputField renders with type password 1`] = `
<Field
  className="stx-field--with-input"
  htmlFor="stxField14"
>
  <Input
    id="stxField14"
    selectOnFocus={false}
    type="password"
  />
</Field>
`;

exports[`InputField InputField renders with type search 1`] = `
<Field
  className="stx-field--with-input"
  htmlFor="stxField9"
>
  <Input
    id="stxField9"
    selectOnFocus={false}
    type="search"
  />
</Field>
`;

exports[`InputField InputField renders with type tel 1`] = `
<Field
  className="stx-field--with-input"
  htmlFor="stxField11"
>
  <Input
    id="stxField11"
    selectOnFocus={false}
    type="tel"
  />
</Field>
`;

exports[`InputField InputField renders with type text 1`] = `
<Field
  className="stx-field--with-input"
  htmlFor="stxField5"
>
  <Input
    id="stxField5"
    selectOnFocus={false}
    type="text"
  />
</Field>
`;

exports[`InputField InputField renders with type time 1`] = `
<Field
  className="stx-field--with-input"
  htmlFor="stxField8"
>
  <Input
    id="stxField8"
    selectOnFocus={false}
    type="time"
  />
</Field>
`;

exports[`InputField InputField renders with type url 1`] = `
<Field
  className="stx-field--with-input"
  htmlFor="stxField13"
>
  <Input
    id="stxField13"
    selectOnFocus={false}
    type="url"
  />
</Field>
`;

exports[`InputField InputField renders without any props 1`] = `
<Field
  className="stx-field--with-input"
  htmlFor="stxField2"
>
  <Input
    id="stxField2"
    selectOnFocus={false}
    type="text"
  />
</Field>
`;
